gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +166,272 @@ def _get_prior_measurements_in_world_frame(
 
         def get_prior_in_world_frame(i2, i2Ti1_prior):
             return wRi_list[i2].rotate(i2Ti1_prior.value.translation())
 
-        w_i2ti1_priors = gtsam.BinaryMeasurementsPoint3()
+        w_i1ti2_prior_measurements = BinaryMeasurementsPoint3()
+        if len(i2Ti1_priors) == 0:
+            return w_i1ti2_prior_measurements
 
+        # TODO(akshay-krishnan): Use the translation covariance, transform to world frame.
+        # noise_model = gtsam.noiseModel.Gaussian.Covariance(i2Ti1_prior.covariance)
+        # TODO(akshay-krishnan): use robust noise model for priors?
+        noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)
         for (i1, i2), i2Ti1_prior in i2Ti1_priors.items():
-            # TODO(akshay-krishnan): Use the translation covariance, transform to world frame.
-            # noise_model = gtsam.noiseModel.Gaussian.Covariance(i2Ti1_prior.covariance)
-            noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)
-            w_i2ti1_priors.append(
+            w_i1ti2_prior_measurements.append(
                 BinaryMeasurementPoint3(
-                    i2,
-                    i1,
+                    C(i2),
+                    C(i1),
                     get_prior_in_world_frame(i2, i2Ti1_prior),
                     noise_model,
                 )
             )
-        return w_i2ti1_priors
+        return w_i1ti2_prior_measurements
+
+    def compute_inliers(
+        self,
+        w_i2Ui1_dict: RelativeDirectionsDict,
+        w_i2Ui1_dict_tracks: RelativeDirectionsDict,
+    ) -> Tuple[RelativeDirectionsDict, RelativeDirectionsDict, Set[int]]:
+        """Perform inlier detection for the relative direction measurements.
+
+        Args:
+            w_i2Ui1_dict: Dictionary of Unit3 relative translations between cameras.
+            w_i2Ui1_dict_tracks: Dictionary of Unit3 relative translations between cameras and landmarks.
+
+        Returns:
+            Tuple of:
+            inlier_w_i2Ui1_dict: Dictionary of inlier Unit3 relative translations between cameras.
+            inlier_w_i2Ui1_dict_tracks: Dictionary of inlier Unit3 relative translations between cameras and landmarks.
+            inlier_cameras: Set of inlier cameras.
+        """
+
+        # Sample directions for projection
+        combined_measurements = list(w_i2Ui1_dict.values()) + list(w_i2Ui1_dict_tracks.values())
+        projection_directions = self.__sample_projection_directions(combined_measurements)
+
+        # Convert to measurements: map indexes to symbols.
+        dummy_noise_model = gtsam.noiseModel.Isotropic.Sigma(3, 1e-2)  # MFAS does not use this.
+        w_i1Ui2_measurements = self._binary_measurements_from_dict(w_i2Ui1_dict, w_i2Ui1_dict_tracks, dummy_noise_model)
 
-    def __get_initial_values(self, wTi_initial: List[Optional[PosePrior]]):
+        # Compute outlier weights using MFAS.
+        # TODO(ayush): parallelize this step.
+        outlier_weights: List[Dict[Tuple[int, int], float]] = []
```

Comment:
Just curious: can we make outlier_weights dict keyed on symbols instead of integers?

Reply:
it is already. I dont know why there is an int type hint. are Symbols ints in python?